hal/mbed_mpu_api.c

```diff
@@ -0,0 +1,149 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "hal/mpu_api.h"
+#include "cmsis.h"
+
+void mbed_mpu_init()
+{
+    mbed_mpu_enable_ram_xn(false);
+}
+
+void mbed_mpu_free()
+{
+    mbed_mpu_enable_ram_xn(false);
+}
+
+void mbed_mpu_enable_ram_xn(bool enable)
+{
+#if MBED_MPU_ENABLED
+    // Flush memory writes before configuring the MPU.
+    __DSB();
+
+    const uint32_t regions = (MPU->TYPE & MPU_TYPE_DREGION_Msk) >> MPU_TYPE_DREGION_Pos;
+
+    // Disable the MCU
+    MPU->CTRL = 0;
+
+    // Reset all mapping
+    for (uint32_t i = 0; i < regions; i++) {
+        ARM_MPU_ClrRegion(i);
+    }
+
+    if (!enable) {
+        return;
+    }
+
+    /*
+     * ARMv6m and ARMv7m memory map:
```

Comment:
Would it make sense to allow platforms override default memory map on hal level?

Reply:
This may be required if there is a target with non-standard memory, but for now there isn't a good reason to.